sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/CommunicationIdentifier.java

```diff
@@ -23,27 +98,40 @@ public static CommunicationIdentifier fromRawId(String rawId) {
             throw new IllegalArgumentException("The parameter [rawId] cannot be null to empty.");
         }
 
-        if (rawId.startsWith("4:")) {
-            return new PhoneNumberIdentifier(rawId.substring("4:".length()));
+        if (rawId.startsWith(PHONE_NUMBER)) {
+            return new PhoneNumberIdentifier(rawId.substring(PHONE_NUMBER.length()));
         }
         final String[] segments = rawId.split(":");
-        if (segments.length < 3) {
+        if (segments.length != 3) {
+            if (segments.length == 2 && rawId.startsWith(BOT_GLOBAL)) {
+                return new MicrosoftBotIdentifier(segments[1], CommunicationCloudEnvironment.PUBLIC, true);
+            }
             return new UnknownIdentifier(rawId);
         }
 
         final String prefix = segments[0] + ":" + segments[1] + ":";
         final String suffix = rawId.substring(prefix.length());
```

Comment:
just reuse the third item, instead of substring the rawId again

Reply:
Updated